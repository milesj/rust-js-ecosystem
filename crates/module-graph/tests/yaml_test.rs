mod utils;

use starbase_sandbox::{assert_snapshot, create_sandbox};
use utils::generate_graph_for_file;

mod yaml {
    use super::*;

    #[test]
    fn object() {
        let sandbox = create_sandbox("yaml");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "object.yaml"));
    }

    #[test]
    fn array() {
        let sandbox = create_sandbox("yaml");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "array.yaml"));
    }

    #[test]
    fn primitive() {
        let sandbox = create_sandbox("yaml");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "primitive.yaml"));
    }
}
