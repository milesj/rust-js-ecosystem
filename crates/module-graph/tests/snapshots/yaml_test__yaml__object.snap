---
source: crates/module-graph/tests/yaml_test.rs
expression: "generate_graph_for_file(sandbox.path(), \"object.yaml\")"
---
{
    1: Module {
        exports: [
            Export {
                kind: Native,
                module_id: None,
                source: None,
                span: None,
                symbols: [
                    ExportedSymbol {
                        kind: Default,
                        symbol_id: None,
                        name: "default",
                    },
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "a",
                    },
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "b",
                    },
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "c",
                    },
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "d",
                    },
                ],
                type_only: false,
            },
        ],
        fragment: None,
        id: 1,
        imports: [],
        package_name: None,
        path: "/root/object.yaml",
        query: None,
        source: YamlModule {
            data: Mapping {
                "a": Number(123),
                "b": String("abc"),
                "c": Bool(true),
                "d": Mapping {
                    "e": Number(456),
                    "f": String("def"),
                    "g": Bool(false),
                },
            },
            source: "a: 123\nb: abc\nc: true\nd:\n  e: 456\n  f: def\n  g: false\n",
        },
    },
}
